os/arch/arm/src/amebad/amebad_serial.c

```diff
@@ -808,7 +786,16 @@ int up_lowgetc(void)
  *  sent character
  *
  ****************************************************************************/
-int up_putc(int ch)
+
+#define MAX_BUFFER_SIZE 1000
+extern bool abort_mode;
+static char buffer[MAX_BUFFER_SIZE];
+static int start = 0;
+static int end = 0;
+sem_t mutex;
+static bool init_mutex = true;
+
+int putc_helper(int ch)
```

Comment:
What's the meaning of `helper` in the name?

Reply:
Initially, I named the function up_putc_helper, and then since up shouldn't be used in the name of this function, I changed it to putc_helper.